### PR TITLE
Institutions

### PR DESCRIPTION
**Pull request by riberman**

### CODE REVIEW

File: .rake_tasks~ [hunk 1]
```diff
@@ -0,0 +1,52 @@
+about
```
Comment (dmarczal) [inline .rake_tasks~]: This file is not necessary. It should not be on the repository.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-[![Maintainability](https://api.codeclimate.com/v1/badges/f26b57e590687b502890/maintainability)](https://codeclimate.com/github/TSIDW5/walgprog/maintainability)
-[![Test Coverage](https://api.codeclimate.com/v1/badges/f26b57e590687b502890/test_coverage)](https://codeclimate.com/github/TSIDW5/walgprog/test_coverage)
-[![Build Status](https://travis-ci.org/TSIDW5/walgprog.svg?branch=master)](https://travis-ci.org/TSIDW5/walgprog)
+[![Maintainability](https://api.codeclimate.com/v1/badges/7e2b5960951159798f2e/maintainability)](https://codeclimate.com/github/riberman/walgprog/maintainability)
```
Comment (dmarczal) [inline README.md:1]: You should not change these lines.

File: app/assets/javascripts/admins/city.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+$(document).on('ready turbolinks:load', function () {
+
+    console.log("Carregou Página");
+    $('#institution_city').attr('disabled', 'true');
```
Comment (dmarczal) [inline app/assets/javascripts/admins/city.js]: This ```$('#institution_city').attr('disabled', 'true'); ``` should be done on html element, not by js.

File: app/controllers/admins/institutions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+class Admins::InstitutionsController < Admins::BaseController
```
Comment (dmarczal) [inline app/controllers/admins/institutions_controller.rb:1]: This controller should be refactored to:
1. Use before action to load states, cities and find the institution
2. The flash messages should use I18n, see the pt-BR.yml, it has the pattern to flash messages.

File: app/views/admins/institutions/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<% provide(:head_title, 'Editar Instituição de Ensino') %>
+<% provide(:page_title, 'Editar Instituição de Ensino') %>
+<% provide(:title, 'Editar Instituição de Ensino') %>
+
+
```
Comment (dmarczal) [inline app/views/admins/institutions/edit.html.erb]: Remove this empty line.

File: app/views/admins/institutions/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+<%= simple_form_for([:admins, @institution]) do |f| %>
+  <%= f.error_notification %>
```
Comment (dmarczal) [inline app/views/admins/institutions/edit.html.erb]: This line is not necessary. the flash message do this.

File: app/views/admins/institutions/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      <%= f.input :name %>
+    </div>
+    <div class="col-lg-4">
+      <%= f.input :acronym %>
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col-lg-4">
+      <%= f.input :state, collection: @states, prompt: 'Escolha o estado', selected: @institution.city.state.id %>
```
Comment (dmarczal) [inline app/views/admins/institutions/edit.html.erb]: You should use I18n for prompt.

File: app/views/admins/institutions/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    </div>
+    <div class="col-lg-8">
+      <%= f.input :city, collection: @cities, prompt: 'Selecione um estado', selected: @institution.city.id, input_html: { name: 'institution[city_id]' }%>
```
Comment (dmarczal) [inline app/views/admins/institutions/edit.html.erb]: This 
```
 selected: @institution.city.id, input_html: { name: 'institution[city_id]' }
```
is not necessary.

File: app/views/admins/institutions/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,51 @@
+<% provide(:head_title, 'Instituições de Ensino') %>
+<% provide(:page_title, 'Instituições de Ensino') %>
+<% provide(:title, 'Instituições de Ensino') %>
+
+<div class="row">
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb:5]: It's not indented.

File: app/views/admins/institutions/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        <tr>
+          <th>Nome</th>
+          <th>Sigla</th>
+          <th>Cidade</th>
+          <th>Estado</th>
+          <th colspan="3"></th>
+        </tr>
+        </thead>
+        <tbody>
+          <% @institutions.each do |institution| %>
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb]: Use ```<%= render @instituctions %>``` and create the right partial

File: app/views/admins/institutions/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+      <div class="col-md-4 col-sm-12">
+        <%= link_to "+ Nova Instituição", new_admins_institution_path, class: "btn btn-outline-primary btn-block"  %>
+      </div>
+    </div>
+    <div class="table-responsive">
+      <table class="table card-table table-striped table-vcenter">
+        <thead>
+        <tr>
+          <th>Nome</th>
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb]: All columns name should be done using I18n.

File: app/views/admins/institutions/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        </tr>
+        </thead>
+        <tbody>
+          <% @institutions.each do |institution| %>
+        <tr>
+          <td><%= institution.name %></td>
+          <td><%= institution.acronym %></td>
+          <td class="text-nowrap"><%= institution.city.name %></td>
+          <td class="text-nowrap"><%= institution.city.state.acronym %></td>
+          <td class="w-1"><%= link_to '<i class="fe fe-edit"></i>'.html_safe, edit_admins_institution_path(institution), :title=> "Editar" %></td>
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb]: Put all links options in the same td.

File: app/views/admins/institutions/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+          <td class="w-1"><%= link_to '<i class="fe fe-trash"></i>'.html_safe, admins_institution_path(institution),
+              method: :delete,
+              data: { confirm: 'Deseja remover?' }, :title=> "Deletar" %></td>
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb]: Use I18n.

File: app/views/admins/institutions/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        </thead>
+        <tbody>
+          <% @institutions.each do |institution| %>
+        <tr>
+          <td><%= institution.name %></td>
+          <td><%= institution.acronym %></td>
+          <td class="text-nowrap"><%= institution.city.name %></td>
+          <td class="text-nowrap"><%= institution.city.state.acronym %></td>
+          <td class="w-1"><%= link_to '<i class="fe fe-edit"></i>'.html_safe, edit_admins_institution_path(institution), :title=> "Editar" %></td>
+          <td class="w-1"><%= link_to '<i class="fe fe-trash"></i>'.html_safe, admins_institution_path(institution),
```
Comment (dmarczal) [inline app/views/admins/institutions/index.html.erb]: This should be done as:
```
<%= link_to admins_institution_path(institution), method: :delete,
             data: { confirm: 'Deseja remover?' },  title: "Deletar" do %>
    <%= icon('fe', 'trash') %>
<% end %>
```

File: app/views/admins/institutions/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,33 @@
+<% provide(:head_title, 'Cadastrar Instituição de Ensino') %>
+<% provide(:page_title, 'Cadastrar Instituição de Ensino') %>
+<% provide(:title, 'Cadastrar Instituição de Ensino') %>
+
+
```
Comment (dmarczal) [inline app/views/admins/institutions/new.html.erb]: Remove the empty line.

File: app/views/admins/institutions/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+<%= simple_form_for([:admins, @institution]) do |f| %>
+  <%= f.error_notification %>
```
Comment (dmarczal) [inline app/views/admins/institutions/new.html.erb]: This line is not necessary. The Flash message do this.

File: app/views/admins/institutions/new.html.erb [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+      <%= f.input :acronym %>
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col-lg-4">
+      <%= f.input :state, collection: @states, prompt: 'Escolha o estado' %>
+    </div>
+    <div class="col-lg-8">
+      <%= f.input :city, collection: [], prompt: 'Selecione uma cidade', input_html: {name: 'institution[city_id]'} %>
```
Comment (dmarczal) [inline app/views/admins/institutions/new.html.erb]: Review as commented in the class.

File: app/views/admins/institutions/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+      <%= f.input :name %>
+    </div>
+    <div class="col-lg-4">
+      <%= f.input :acronym %>
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col-lg-4">
+      <%= f.input :state, collection: @states, prompt: 'Escolha o estado' %>
```
Comment (dmarczal) [inline app/views/admins/institutions/new.html.erb]: Use I18n.

File: app/views/layouts/admins/_sidebar.html.erb [hunk 19]
```diff
@@ -11,5 +11,9 @@
     <%= link_to admins_root_path, class: 'list-group-item list-group-item-action', active: :exclusive do %>
       <span class="icon mr-3"><i class="fas fa-home"></i></span>Página inicial
     <% end %>
+
+    <%= link_to admins_institutions_path, class: 'list-group-item list-group-item-action', active: :exclusive do %>
+      <span class="icon mr-3"><i class="fas fa-university"></i></span>Instituições de Ensino
```
Comment (dmarczal) [inline app/views/layouts/admins/_sidebar.html.erb]: If icon is from fontawesome use  icon function ```<%= icon('fas', 'user-cog', class: 'mr-2') %>```

File: app/views/layouts/admins/application.html.erb [hunk 20]
```diff
@@ -45,5 +44,6 @@
       </div>
       <%= render 'layouts/admins/footer' %>
     </div>
+        <%= javascript_include_tag 'admins/application', 'data-turbolinks-track': 'reload' %>
```
Comment (dmarczal) [inline app/views/layouts/admins/application.html.erb]: This is in the wrong place.

File: spec/features/admins/institutions/admin_new_institution_spec.rb [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+    fill_in 'institution_name', with: 'Universidade Teste'
+    fill_in 'institution_acronym', with: 'UT'
+    state = State.first
+    select(state.name, from: 'institution_state')
+    select(state.cities.sample.name, from: 'institution_city')
+
+    click_button
+
+    expect(page).to have_current_path admins_institutions_path
+    # expect(page).to have_selector('div.alert.alert-success',
```
Comment (dmarczal) [inline spec/features/admins/institutions/admin_new_institution_spec.rb]: You should expect a flash message.

File: spec/spec_helper.rb [hunk 22]
```diff
@@ -1,13 +1,96 @@
-require 'support/simplecov'
-
+# This file was generated by the `rails generate rspec:install` command. Conventionally, all
```
Comment (dmarczal) [inline spec/spec_helper.rb]: Why did you add these comments? Remove them.

Why did you remove this line require 'support/simplecov' ? to pass at Travis?

**Review by dmarczal — CHANGES_REQUESTED** [review]
1. Use Selectize for selects. Se examples in tabler.io.
2. Sort the institutions by name Asc.
3. Create the populate. Use [this](https://github.com/utfpr-gp-tsi/SGTCC/blob/master/lib/tasks/populate.rake) as an example.
4. Get the news features from the master. Take care of the conflicts

File: app/assets/javascripts/admins/city.js [hunk 23]
```diff
@@ -0,0 +1,23 @@
+$(document).on('ready turbolinks:load', function () {
+
+    console.log("Carregou Página");
```
Comment (pedroida) [inline app/assets/javascripts/admins/city.js]: remover console.log()
